### ISSUE TITLE
(WCAG 2.1) Close control is not keyboard accessible

### ISSUE THREAD

**Issue by dkatzz** [civiform/civiform#5521]
OBSERVED RESULTS:

- The Close control on a “Create an account or sign in” modal dialog is not keyboard accessible and cannot be operated using keyboard navigation.
- ![Screenshot](https://github.com/civiform/civiform/assets/86739416/dbe7718e-4315-493e-a98f-2303b274076e)

EXPECTED RESULTS:

- The Close control on a “Create an account or sign in” modal dialog should be fully keyboard accessible, allowing users to close the modal using the keyboard.
- Use a role button and or use <button> to implement the close button.
- Provide valid label using aria-label to the close button.

STEPS TO REPRODUCE:

- Navigate to [URL](https://demo.civiform.dev/)
- Navigate to and activate the ‘Apply’ for the Comprehensive sample program.
- Navigate to and activate “What is your address?”.
- Fillup the form fields and complete the application.
- On the confirmation page navigate to and activate “Apply to Another Program" using the tab key.
- Navigate to the “close” button using TAB key and observe that the control is not keyboard accessible.

ENVIRONMENT DETAILS:

- Device(s) (brand, OS version, test key or release key): Windows 10 Enterprise (64-bit),MacOS Ventura Version 13.4
- Screen reader name and version: JAWS 2022.2211.7, NVDA 2022.3, VoiceOver
- Browser name and version: Chrome Version 113.0.5672.93

**Comment by caitlinshk** [comment]
Note: The close button on the question bank when adding questions to a program also isn't accessible, so we should try and fix both.
![4xViVtAts4z67gW](https://github.com/civiform/civiform/assets/147177873/26a8de48-2c24-4d64-bb98-cba99c114255)

**Comment by melanie-exygy** [comment]
> Note: The close button on the question bank when adding questions to a program also isn't accessible, so we should try and fix both. 

We haven't prioritized accessibility for the admin side of the app, so making the close button accessible on the "Add a question" sidebar is probably less important.  Still, if we can fix it easily, it seems like a good thing to do!

**Comment by caitlinshk** [comment]
Ah that makes sense, thanks for the clarification!
